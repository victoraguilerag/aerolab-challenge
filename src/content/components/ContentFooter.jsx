import React from 'react'
import PropTypes from 'prop-types'
import previusIcon from '../../icons/arrow-left.svg'
import nextIcon from '../../icons/arrow-right.svg'

ContentFooter.propTypes = {
  prev: PropTypes.bool.isRequired,
  next: PropTypes.bool.isRequired,
  handlePreviousPage: PropTypes.func.isRequired,
  handleNextPage: PropTypes.func.isRequired,
  pageLength: PropTypes.number.isRequired,
  productsLength: PropTypes.number.isRequired
}

function ContentFooter (props) {
  const {
    prev,
    handlePreviousPage,
    next,
    handleNextPage,
    pageLength,
    productsLength
  } = props
  return (
    <div className="content-footer" style={styles.contentFooter}>
      <p className="products-count-footer" style={styles.productCount}>
        {`${pageLength} of ${productsLength} items`}
      </p>
      <div className="products-controls" style={styles.productsControls}>
        {
          prev &&
          <img src={previusIcon} style={styles.control} width="46px" height="46px" className="previus" alt="previus" onClick={handlePreviousPage} />
        }
        {
          next &&
          <img src={nextIcon} style={styles.control} width="46px" height="46px" className="next" alt="next" onClick={handleNextPage} />
        }
      </div>
    </div>
  )
}

const styles = {
  'contentFooter': {
    'display': 'flex',
    'justifyContent': 'flex-start',
    'alignItems': 'center',
    'margin': '0 auto',
    'height': '70px',
    'maxWidth': '1176px',
    'borderBottom': 'solid 1px #d9d9d9',
    'marginTop': '59px',
    'marginBottom': '100px',
    'position': 'relative'
  },
  'productCount': {
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'left',
    'margin': '0 10px'
  },
  'productsControls': {
    'position': 'absolute',
    'right': '5px'
  },
  'control': {
    'margin': '0 10px',
    'cursor': 'pointer'
  }
}

export default ContentFooter
