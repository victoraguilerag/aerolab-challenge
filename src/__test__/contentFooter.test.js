import React from 'react'
import ContentFooter from '../content/components/ContentFooter.jsx'
import renderer from 'react-test-renderer'

const filters = []
const functionMock = () => console.log('mock')
const pageLength = 2
const productsLength = 16

test('Content Options renders', () => {
  const component = renderer.create(
    <ContentFooter
      prev={false}
      next={true}
      handlePreviousPage={functionMock}
      handleNextPage={functionMock}
      pageLength={pageLength}
      productsLength={productsLength}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
