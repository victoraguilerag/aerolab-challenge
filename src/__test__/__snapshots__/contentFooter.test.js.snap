// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Options renders 1`] = `
<div
  className="content-footer"
  style={
    Object {
      "alignItems": "center",
      "borderBottom": "solid 1px #d9d9d9",
      "display": "flex",
      "height": "70px",
      "justifyContent": "flex-start",
      "margin": "0 auto",
      "marginBottom": "100px",
      "marginTop": "59px",
      "maxWidth": "1176px",
      "position": "relative",
    }
  }
>
  <p
    className="products-count-footer"
    style={
      Object {
        "alignItems": "center",
        "color": "#616161",
        "display": "flex",
        "fontSize": "24px",
        "justifyContent": "center",
        "letterSpacing": "-0.15px",
        "lineHeight": "48px",
        "margin": "0 10px",
        "textAlign": "left",
      }
    }
  >
    2 of 16 items
  </p>
  <div
    className="products-controls"
    style={
      Object {
        "position": "absolute",
        "right": "5px",
      }
    }
  >
    <img
      alt="next"
      className="next"
      height="46px"
      onClick={[Function]}
      src={Object {}}
      style={
        Object {
          "cursor": "pointer",
          "margin": "0 10px",
        }
      }
      width="46px"
    />
  </div>
</div>
`;
